### DISCUSSION TITLE
Passing "uint256[] memory" type arguments to function. in Remix

### DISCUSSION THREAD

**Discussion by msint** [smartcontractkit/full-blockchain-solidity-course-py#1331] [Q&A] [answered]

**Comment by cromewar** [comment, marked as answer]
Hello @msint 
Did you tried just declaring the array before send it?
Instead of use the actual values on the function call you can try sending the whole array as a parameter.
Also if you share your code here it'll be easier to help you.
Cheers.

**Reply by n4n0b1t3** [reply]
you wrote
> I don't know how to pass the function arguments with type "uint256[] memory" 

There is no need to pass anything to fulfillRandomWords, because it is a callback function which will be automatically called from requestRandomness. Since you are inheriting from VRFConsumerBaseV2 you only need to call  
`bytes32 requestId = requestRandomness(keyhash, fee)`

This calls ...
```python
function fulfillRandomWords(
    uint256, /* requestId */
    uint256[] memory randomWords
  ) internal override {
    s_randomWords = randomWords; 
}
```
After requestRandomness is called succesfully, with the above code s_randomWords would already contain a very long number.

[try in remix](https://remix.ethereum.org/#url=https://docs.chain.link/samples/VRF/VRFv2Consumer.sol)

**Reply by cromewar** [reply]
Does @n4n0b1t3 make sense for you @msint ?
